tests/PSRule.Tests/PSRule.Common.Tests.ps1

```diff
@@ -1125,6 +1125,119 @@ Describe 'Invoke-PSRule' -Tag 'Invoke-PSRule','Common' {
             $outVerbose[1] | Should -Be 'Rule verbose message 2';
         }
     }
+
+    Context 'Suppression output warnings' {
+        BeforeAll {
+            $testObject = @(
+                [PSCustomObject]@{
+                    Name = "TestObject1"
+                }
+                [PSCustomObject]@{
+                    Name = "TestObject2"
+                }
+            )
+        }
+
+        Context 'Detail' {
+            It 'From Parameter - Has warnings' {
+                $option = New-PSRuleOption -SuppressTargetName @{ FromFile1 = 'TestObject1'; FromFile2 = 'TestObject1'; } -SuppressedRuleWarning $True -OutputAs Detail;
+
+                $Null = $testObject | Invoke-PSRule -Path $ruleFilePath -Option $option -Name 'FromFile1', 'FromFile2' -WarningVariable outWarnings -WarningAction SilentlyContinue;
+    
+                $warningMessages = $outwarnings.ToArray();
+                $warningMessages.Length | Should -Be 2;
+    
+                $warningMessages[0] | Should -BeOfType [System.Management.Automation.WarningRecord];
+                $warningMessages[0].Message | Should -BeExactly "Rule 'FromFile1' was suppressed for 'TestObject1'.";
+                $warningMessages[1] | Should -BeOfType [System.Management.Automation.WarningRecord];
+                $warningMessages[1].Message | Should -BeExactly "Rule 'FromFile2' was suppressed for 'TestObject1'.";
+            }
+
+            It 'From Parameter - No warnings' {
+                $option = New-PSRuleOption -SuppressTargetName @{ FromFile1 = 'TestObject1'; FromFile2 = 'TestObject1'; } -SuppressedRuleWarning $False -OutputAs Detail;
+
+                $Null = $testObject | Invoke-PSRule -Path $ruleFilePath -Option $option -Name 'FromFile1', 'FromFile2' -WarningVariable outWarnings -WarningAction SilentlyContinue;
+    
+                $warningMessages = $outwarnings.ToArray();
+                $warningMessages.Length | Should -Be 0;
+            }
+
+            It 'From Yaml - Has warnings' {
```

Comment:
I don't think it is necessary to add these additional YAML tests separate from the option (choose one) because once the option is loaded as an option object it's the same code path. We already tested the option object is correct in the latest tests.

Reply:
Yeah agreed, will remove the Yaml tests.